opentracing-api/src/main/java/io/opentracing/ScopeManager.java

```diff
@@ -58,12 +59,43 @@
      *     }
      * </code></pre>
      *
+     * <p>
+     * Note: You may only activate spans when you own its life cycle.
+     * That means you must make sure that no other thread calls {@link Span#finish()}
+     * while the scope is still active.
+     * If you can't guarantee that, use {@link #activate(SpanContext)} instead.
+     *
      * @param span the {@link Span} that should become the {@link #activeSpan()}
      * @return a {@link Scope} instance to control the end of the active period for the {@link Span}. It is a
      * programming error to neglect to call {@link Scope#close()} on the returned instance.
      */
     Scope activate(Span span);
 
+    /**
+     * Similar to {@link #activate(Span)} but used in cases where the thread in which the
+     * activation is performed does not have control over the life cycle of the span.
+     *
+     * <p>
+     * One example of that is when performing an activation in the {@link Runnable#run()}
+     * method of a traced {@link Runnable} wrapper which is executed by an
+     * {@link java.util.concurrent.ExecutorService}.
+     *
```

Comment:
add example in code? So far doesn't explicitly mention try/finally. I think this will reinforce it

Reply:
added an example and copied over the try-with-resources paragraph